js/ui/handler/touch_zoom_rotate.js

```diff
@@ -55,9 +62,30 @@ TouchZoomRotate.prototype = {
             bearing = this._rotationDisabled ? 0 : vec.angleWith(this._startVec) * 180 / Math.PI,
             map = this._map;
 
+        this._scalingSignificantly =
+            this._scalingSignificantly || (Math.abs(1 - scale) > this._significantScaleThreshold);
+
+        this._rotatingSignificantly =
+            this._rotatingSignificantly || (Math.abs(bearing) > this._significantRotateThreshold);
+
+        // Similar to google maps: if the user's intent is pinch zooming, rotate
+        // will be disabled. If their intent is rotating, then both rotate and
+        // zoom will be supported. We determine 'intent' by whichever threshold is
+        // surpassed first. Once determined, we keep that state for the duration
+        // of this gesture.
+        if (this._blockRotation === undefined) {
+            if (this._scalingSignificantly) {
+                this._blockRotation = true;
+            } else if (this._rotatingSignificantly) {
+                this._blockRotation = false;
+            }
+        }
+
+        // Only set the bearing if rotation is allowed AND we are rotating more than our threshold
+        var updateBearing = ((this._blockRotation === false) && this._rotatingSignificantly);
```

Comment:
So, I noticed that other native maps seem to lock in the intent first and then allow the movement.  The way you are calling `easeTo` lets a little zoom movement sneak in before it's clear what the user wants.  Anyway, this is what I did - try it and let me know what you think:
```javascript
        var updateBearing = ((this._blockRotation === false) && this._rotatingSignificantly),
            param = {duration: 0, around: map.unproject(p)};

        if (updateBearing) {
            param.bearing = this._startBearing + bearing;
        }
        if (this._blockRotation !== undefined) {
            param.zoom = map.transform.scaleZoom(this._startScale * scale);
        }

        map.easeTo(param);
```